storage/local.go

```diff
@@ -16,21 +18,77 @@ import (
 // copying the VSIX and extracting said VSIX to a tree structure in the form of
 // publisher/extension/version to easily serve individual assets via HTTP.
 type Local struct {
-	extdir string
-	logger slog.Logger
+	listCache      []extension
+	listDuration   time.Duration
+	listExpiration time.Time
+	listMutex      sync.Mutex
+	extdir         string
+	logger         slog.Logger
 }
 
-func NewLocalStorage(extdir string, logger slog.Logger) (*Local, error) {
-	extdir, err := filepath.Abs(extdir)
+type LocalOptions struct {
+	// How long to cache list responses.  Zero means no cache.  Manifests are
+	// currently cached indefinitely since they do not change.
```

Comment:
The Artifactory implementation has a separate manifest cache on top of the list cache that never expires but it looks like in the local implementation we have the one cache and it has manifests in it so we should update the comment to match.  Maybe something like:

```suggestion
	// How long to cache the list of extensions and their manifests.  Zero means 
	// no cache.
```

Reply:
Yes, that sounds like a good idea. I initially experimented with making a separate cache for manifest in my testing because I figured the parsing of XML-files would be a bottleneck, but it turned out to be negligible compared to the file traversal (like <5% of the total processing time), so I removed it.

Reply:
Makes sense!